fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java

```diff
@@ -294,8 +221,466 @@ public PhysicalRelation visitPhysicalScan(PhysicalRelation scan, CascadesContext
+                    EqualTo equalTo = (EqualTo) join.getHashJoinConjuncts().get(0);
+                    equalTos.add(equalTo);
+                    equalCondToJoinMap.put(equalTo, join);
+                }
+            }
+            if (joinSet.size() == equalTos.size()) {
+                int matchNum = 0;
+                Set<String> cteNameSet = new HashSet<>();
+                Set<SlotReference> anotherSideSlotSet = new HashSet<>();
+                for (EqualTo equalTo : equalTos) {
+                    // has been checked above, SlotReference for both sides
+                    SlotReference left = (SlotReference) equalTo.left();
+                    SlotReference right = (SlotReference) equalTo.right();
+                    if (left.getQualifier().size() == 1 && left.getQualifier().get(0).equals(cteName)) {
+                        matchNum += 1;
+                        anotherSideSlotSet.add(right);
+                        cteNameSet.add(left.getQualifiedName());
+                    } else if (right.getQualifier().size() == 1 && right.getQualifier().get(0).equals(cteName)) {
+                        matchNum += 1;
+                        anotherSideSlotSet.add(left);
+                        cteNameSet.add(right.getQualifiedName());
+                    }
+                }
+                if (matchNum == equalTos.size() && cteNameSet.size() == 1) {
+                    // means all join condition points to the same cte on the same cte column.
+                    // collect the other side columns besides cte column side.
+                    Preconditions.checkState(equalTos.size() == equalCondToJoinMap.size());
+
+                    PhysicalCTEProducer cteProducer = context.getRuntimeFilterContext().getCteProduceMap().get(cteId);
+                    if (anotherSideSlotSet.size() == 1) {
+                        // meet requirement for pushing down into cte producer
+                        ctx.getCteRFPushDownMap().put(cteProducer, equalCondToJoinMap);
+                    } else {
+                        // check further whether the join upper side can bring equal set, which
+                        // indicating actually the same runtime filter build side
+                        List<Expression> conditions = curJoin.getHashJoinConjuncts();
+                        boolean inSameEqualSet = false;
+                        for (Expression e : conditions) {
+                            if (e instanceof EqualTo) {
+                                SlotReference oneSide = (SlotReference) ((EqualTo) e).left();
+                                SlotReference anotherSide = (SlotReference) ((EqualTo) e).right();
+                                if (anotherSideSlotSet.contains(oneSide) && anotherSideSlotSet.contains(anotherSide)) {
+                                    inSameEqualSet = true;
+                                    break;
+                                }
+                            }
+                        }
```

Comment:
cannot understand easily, could u offer an example in comment?

Reply:
done